src/Router/Zf2.php

```diff
@@ -109,63 +155,17 @@ private function createRouter()
      */
     private function marshalSuccessResultFromRouteMatch(RouteMatch $match)
     {
-        $params = $match->getParams();
-        $middleware = isset($params['middleware'])
-            ? $params['middleware']
-            : $this->getMiddlewareFromRoute($match->getMatchedRouteName());
+        $params    = $match->getParams();
+        $routeName = $match->getMatchedRouteName();
+
+        if (null === $params['middleware']) {
+            return RouteResult::fromRouteFailure($this->routes[$routeName]);
+        }
```

Comment:
Wouldn't this indicate a 405 situation? In which case, we should be passing the allowed methods…